### PR TITLE
205 fix so program waits when you are out of token requests

### PR DESCRIPTION
**Pull request by KarlssonPaGolvet**
Fixes so that the program no longer crashes when out of token requests. It now waits until token has reset and the continuous

### CODE REVIEW

File: src/main/util.py [hunk 1]
```diff
@@ -81,14 +108,17 @@ def find_matching_release(release_tags: list[str], version: str) \
...
+
         # Convert the reset time to a human-readable format
         reset_time = datetime.datetime.fromtimestamp(int(reset_time))
         reset_time = reset_time.strftime("%Y-%m-%d %H:%M:%S")
-
-        raise ConnectionRefusedError(
-            (f"GitHub API rate limit exceeded. Try again later."
-             f"Rate limit resets at {reset_time}.")
-            )
+        print(f"GitHub API rate limit exceeded. Rate limit resets at {reset_time}.")
```
Comment (edvgi966) [inline src/main/util.py:116]: Perhaps not a print statement this deep down, could ruin something when you want to put it in a pipeline

Comment (KarlssonPaGolvet) [reply src/main/util.py:116]: Is it better to just leave it going? Thought it might be confusing, but can remove the print statement ^^

Comment (KarlssonPaGolvet) [reply src/main/util.py]: Removed print statement now ^^

**Comment by edvgi966** [comment]
Currently testing, will review once the hour of waiting has passed

**Review by edvgi966 — APPROVED** [review]
Hard to test, but I let it run for a while and seems to work

File: src/main/data_types/dependency_scorer.py [hunk 2]
```diff
@@ -391,8 +392,16 @@ def _execute_scorecard(self,
...
                     timeout=timeout
                 ).decode("utf-8")
-        except subprocess.CalledProcessError as e:
+        except (subprocess.CalledProcessError) as e:
             output = e.output.decode("utf-8")
+        except subprocess.TimeoutExpired as e:
+            token_data: dict = get_token_data()
+            if token_data.get("remaining") > 0:
+                raise e
+            sleep(token_data.get("time_until_reset") + 10)
```
Comment (grufe) [inline src/main/data_types/dependency_scorer.py:401]: Would it be possible to send a StepResponse here to notify of sleep?

**Review by grufe — CHANGES_REQUESTED** [review]

**Review by grufe — DISMISSED** [review]

**Pull request merged**